protocol/messenger_saved_address.go

```diff
@@ -32,6 +35,45 @@ func (m *Messenger) garbageCollectRemovedSavedAddresses() error {
 	return m.savedAddressesManager.DeleteSoftRemovedSavedAddresses(uint64(time.Now().AddDate(0, 0, -30).Unix()))
 }
 
+func (m *Messenger) setInstallationHostname() error {
+	randomDeviceIDLen := 5
+
+	ourInstallation, ok := m.allInstallations.Load(m.installationID)
+	if !ok {
+		m.logger.Error("Messenger's installationID is not set or not loadable")
+		return nil
+	}
+
+	var imd *multidevice.InstallationMetadata
+	if ourInstallation.InstallationMetadata == nil {
+		imd = new(multidevice.InstallationMetadata)
+	} else {
+		imd = ourInstallation.InstallationMetadata
+	}
+
+	// If the name is already set, don't do anything
+	// TODO check the full working mechanics of this
+	if len(imd.Name) > randomDeviceIDLen {
+		return nil
+	}
+
```

Comment:
@cammellos Do you know if a default value is set? In the current UI we show a 5 character random string, but I don't see where that comes from.

Is this the last/first 5 chars of the installation ID?

If so I can remove all the random string appending in the `InstallationMetadata.Name`

Reply:
> Is this the last/first 5 chars of the installation ID?

yes, you can remove the random thing from here, it's take from installation id